### DISCUSSION TITLE
Asset link not rendered after SSG

### DISCUSSION THREAD

**Discussion by davidedistaso** [statamic/cms#4478] [Troubleshooting] [answered]
I deploy Statamic using SSG on Netlify.

I wanted to share a link to a PDF on my website. I created an asset container using the "public" disk and uploaded my pdf there.

`'public' => [
            'driver' => 'local',
            'root' => storage_path('app/public'),
            'url' => env('APP_URL').'/storage',
            'visibility' => 'public',
        ],`

I then created a blueprint with an asset field type which uses the newly made container.

In my template I use `{{ assets:url }}` to show a link to the file (which is a PDF). Everything works locally, but when I deploy to Netlify `{{ assets:url }}` doesn't render the link to the asset. If I go the asset's address, I get a 404 error. I already tried `php artisan storage:link` but nothing changes.

**Comment by duncanmcclean** [comment, marked as answer]
I think you'll need to add a 'copy' to the SSG config (`config/statamic/ssg.php`) to copy files from `storage/app/public` to `storage` to make that work.

**Reply by davidedistaso** [reply]
Tried redeploying after adding this to `config/statamic/ssg.php`
`'copy' => [
        public_path('css') => 'css',
        public_path('js') => 'js',
        storage_path('app/public') => 'storage',   
    ],`

Also tried

`'copy' => [
        public_path('css') => 'css',
        public_path('js') => 'js',
        'storage/app/public' => 'storage',
    ],`

But no difference, is that what you suggested doing?

**Reply by duncanmcclean** [reply]
Is `storage/app/public` included in version control? If not, it'll have nothing to copy. 

I'd recommend using the [assets](https://github.com/statamic/statamic/blob/master/config/filesystems.php#L69) filesystem that ships with a default Statamic install to store your PDF. It'll mean your PDF will live in `public/assets`. Be sure to update the disk in your asset container and in the SSG config.

**Reply by davidedistaso** [reply]
Forgot to delete .gitignore, works perfectly, thank you so much!
